pkg/controllers/execution/execution_controller.go

```diff
@@ -151,16 +146,36 @@ func (c *Controller) syncWork(ctx context.Context, clusterName string, work *wor
 	metrics.ObserveSyncWorkloadLatency(err, start)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to sync work(%s/%s) to cluster(%s), err: %v", work.Namespace, work.Name, clusterName, err)
-		klog.Errorf(msg)
+		klog.Error(msg)
 		c.EventRecorder.Event(work, corev1.EventTypeWarning, events.EventReasonSyncWorkloadFailed, msg)
 		return controllerruntime.Result{}, err
 	}
 	msg := fmt.Sprintf("Sync work(%s/%s) to cluster(%s) successful.", work.Namespace, work.Name, clusterName)
-	klog.V(4).Infof(msg)
+	klog.V(4).Info(msg)
 	c.EventRecorder.Event(work, corev1.EventTypeNormal, events.EventReasonSyncWorkloadSucceed, msg)
 	return controllerruntime.Result{}, nil
 }
 
+func (c *Controller) handleWorkDelete(ctx context.Context, work *workv1alpha1.Work, cluster *clusterv1alpha1.Cluster) error {
+	if ptr.Deref(work.Spec.PreserveResourcesOnDeletion, false) {
+		klog.V(4).Infof("Preserving resource on deletion from work(%s/%s) on cluster(%s)", work.Namespace, work.Name, cluster.Name)
```

Comment:
Should we delete labels and annotations related to resources in the member cluster?

Reply:
is there an existing utility function that does this? I only saw one that updates labels/annotations on ResourceTemplate

Reply:
There is no dedicated function for processing tags and annotations. We may need to add a tool function.

Reply:
Hi @a7i, kindly ping you~ ^-^ do you have time to continue to push the current pr?

Reply:
Do you have any feedback on how to do this in a maintainable way?

This is sort of what I have, but as you can imagine when new labels and annotations get added, we have to make sure that we update this block and it's so easy to miss

```
func (c *Controller) removeAnnotationsAndLabelsInMemberCluster(ctx context.Context, work *workv1alpha1.Work, cluster *clusterv1alpha1.Cluster) error {
       for _, manifest := range work.Spec.Workload.Manifests {
               c.ObjectWatcher.Update()

               fedKey, err := keys.FederatedKeyFunc(clusterName, workload)
               if err != nil {
                       klog.Errorf("Failed to get FederatedKey %s, error: %v", workload.GetName(), err)
                       return err
               }

               clusterObj, err := helper.GetObjectFromCache(c.RESTMapper, c.InformerManager, fedKey)
               if err != nil {
                       klog.Errorf("Failed to get the resource(kind=%s, %s/%s) from member cluster(%s), err is %v ", workload.GetKind(), work.GetNamespace(), workload.GetName(), clusterName, err)
                       return err
               }

               workload := &unstructured.Unstructured{}
               if err := workload.UnmarshalJSON(manifest.Raw); err != nil {
                       klog.Errorf("Failed to unmarshal workload, error is: %v", err)
                       return err
               }

               util.RemoveAnnotations(
                       workload,
                       policyv1alpha1.ClusterPropagationPolicyAnnotation,
                       policyv1alpha1.PropagationPolicyNameAnnotation,
               )

               util.RemoveLabels(
                       workload,
                       policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel,
                       policyv1alpha1.PropagationPolicyPermanentIDLabel,
                       util.ManagedByKarmadaLabel,
               )

               if err := c.ObjectWatcher.Update(ctx, cluster.Name, workload, clusterObj); err != nil {
                       klog.Errorf("Failed to delete resource in the given member cluster %v, err is %v", clusterName, err)
                       return err
               }
       }

       return nil
}
```

Reply:
It does seem a little difficult to maintain.

How about maintain like this: https://github.com/karmada-io/karmada/pull/5563/files#diff-c99707dd17b04d899384bcf56c0f960362d5edcd2f588ebf5ae63d7239e30a3dR27-R39 in pr #5563

Reply:
That's perfect! I will rebase/add as soon as it's merged

Reply:
Hi @a7i, #5563 has been merged.